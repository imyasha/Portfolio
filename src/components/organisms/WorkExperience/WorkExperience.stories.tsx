import { mockJobs } from 'MockData';
import { WorkExperience, WorkExperienceProps } from './WorkExperience';

export default {
	title: 'organisms/WorkExperience',
	component: WorkExperience,
};

export const WorkExperienceStory = (args: WorkExperienceProps) => (
	<WorkExperience {...args} />
);

WorkExperienceStory.storyName = 'WorkExperience';
WorkExperienceStory.args = {
	jobs: mockJobs,
};
