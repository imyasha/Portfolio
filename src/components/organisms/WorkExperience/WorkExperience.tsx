import { IJob } from '@types';

import classNames from 'classnames';
import { RichText } from '@graphcms/rich-text-react-renderer';

import { Job } from 'Molecules/Job';
import { mdxComponents } from 'utils/mdxComponents';

export interface WorkExperienceProps {
	jobs: IJob[];
}

const WorkExperience = ({ jobs }: WorkExperienceProps) => {
	return (
		<>
			{jobs.map(
				(
					{ company, jobTitle, fromDate, toDate, technologies, description },
					index
				) => (
					<div
						className={classNames(
							'flex transition-all transform md:hover:scale-[1.01]',
							{
								'justify-end': index % 2 !== 0,
							}
						)}
						key={`${company} ${jobTitle}`}
					>
						<Job
							company={company}
							jobTitle={jobTitle}
							fromDate={fromDate}
							toDate={toDate}
							technologies={technologies}
						>
							<RichText renderers={mdxComponents} content={description} />
						</Job>
					</div>
				)
			)}
		</>
	);
};

export { WorkExperience };
