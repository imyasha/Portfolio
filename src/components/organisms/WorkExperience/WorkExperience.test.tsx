import { mockJobs } from 'MockData';
import { render } from '@testing-library/react';
import { WorkExperience } from '../WorkExperience';

describe('WorkExperience', () => {
	it('renders correctly', () => {
		const { container } = render(<WorkExperience jobs={[mockJobs[0]]} />);
		expect(container).toMatchSnapshot();
	});
});
