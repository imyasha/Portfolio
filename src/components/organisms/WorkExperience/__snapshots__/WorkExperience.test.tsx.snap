// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkExperience renders correctly 1`] = `
<div>
  <div
    class="flex transition-all transform md:hover:scale-[1.01]"
  >
    <div
      class="md:w-10/12 px-8 py-6 rounded-3xl glass-light dark:glass-dark mb-6 drop-shadow-lg hover:drop-shadow-2xl transition-all"
    >
      <div
        class="flex gap-4 items-start md:items-center"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2764%27%20height=%2764%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Holland & Barrett"
            class="rounded-md"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="w-full mb-2"
        >
          <div
            class="flex flex-col-reverse md:flex-row justify-between"
          >
            <h3
              class="font-bold"
            >
              Senior Software Engineer
            </h3>
            <span
              class="text-sm md:text-base"
            >
              Aug 2020 â€” present
            </span>
          </div>
          <h4>
            <a
              class="underlined relative"
              href="https://www.hollandandbarrett.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Holland & Barrett
            </a>
          </h4>
          <p
            class="hidden md:block text-sm"
          >
            <strong>
              Technologies: 
            </strong>
            TypeScript, React, Next.js, Redux, GraphQL, REST API, Jest, Cypress
          </p>
        </div>
      </div>
      <p
        class="md:hidden text-sm"
      >
        <strong>
          Technologies: 
        </strong>
        TypeScript, React, Next.js, Redux, GraphQL, REST API, Jest, Cypress
      </p>
    </div>
  </div>
</div>
`;
